### ISSUE TITLE
Cannot read fonts included by default on OS X (like Geeza Pro)

### ISSUE THREAD

**Issue by deepakjois** [simoncozens/sile#193]
code:

```
\begin[papersize=a5]{document}
\font[family=Geeza Pro,language=urd,size=36pt]
\center{لکیریں}
\end{document}
```

outputs:

```
This is SILE 0.9.4-unreleased
<geeza.sil>

! Font 'Geeza Pro' not available, falling back to 'Times'
[1]
```

I am pretty sure Geeza Pro is included in OS X by default for a long time. I am having similar problems with some other fonts as well (like Baghdad), so it is not just this font.

**Comment by simoncozens** [comment]
Confirmed. It's a fontconfig bug. I am in the process of getting rid of fontconfig on OS X and replacing it with the OS's font management system, because it causes all kinds of inconvenience. For the time being, if you want to use Geeza Pro, you have to refer to it like this:

     \font[family=جيزة]

And Baghdad:

     \font[family=بغداد]

**Comment by deepakjois** [comment]
Interesting. Arabic names for fonts!

Just curious, how did you find the name of the font. Is there a tool of some kind that you are using to look inside these fonts?

**Comment by simoncozens** [comment]
I used `fc-list`. I now have the most part of an OS X replacement for fontconfig working; will check it in soon, once I get my head around how to correctly fit it into autoconf...

**Comment by simoncozens** [comment]
Please try with the latest HEAD - you will need to completely rebuild (`sh bootstrap.sh ; ./configure ; make`) but it should use Mac OS X's `NSFontManager` class to locate fonts.)

**Comment by deepakjois** [comment]
Getting the same error. I followed steps you specified. Even did a `make clean` and then a `make install`, before running SILE again. Your suggestion to use the Arabic name of the font still works.

**Comment by simoncozens** [comment]
OK; thanks for trying. Could you update again, re-bootstrap and run `./configure` and tell me in particular if you see this:

    checking dependency style of gcc... gcc3
    checking for OS X... yes
    checking for AppKit works... yes

If it does that, it *should* be trying to build `macfonts.m` (have a look through the output of make and see if it does), and if it does *that*, then it should try to use that when configuring fonts.

**Comment by deepakjois** [comment]
Yes, I can see it:

```
/bin/sh ../libtool    --mode=compile gcc -DPACKAGE_NAME=\"SILE\" -DPACKAGE_TARNAME=\"sile\" -DPACKAGE_VERSION=\"0.9.3\" -DPACKAGE_STRING=\"SILE\ 0.9.3\" -DPACKAGE_BUGREPORT=\"simon@simon-cozens.org\" -DPACKAGE_U
RL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1
 -DLT_OBJDIR=\".libs/\" -DAPPKIT=1 -DHAVE_LUA_H=1 -DHAVE_LUALIB_H=1 -DHAVE_LAUXLIB_H=1 -DHAVE_LUACONF_H=1 -DSILE_PATH=\"/usr/local/share/sile\" -DSILE_LIB_PATH=\"/usr/local/lib/sile\" -I.    -I/usr/local/Cellar/
freetype/2.6_1/include/freetype2   -fmodules -g -O2 -MT macfonts_la-macfonts.lo -MD -MP -MF .deps/macfonts_la-macfonts.Tpo -c -o macfonts_la-macfonts.lo `test -f 'macfonts.m' || echo './'`macfonts.m
libtool: compile:  gcc -DPACKAGE_NAME=\"SILE\" -DPACKAGE_TARNAME=\"sile\" -DPACKAGE_VERSION=\"0.9.3\" "-DPACKAGE_STRING=\"SILE 0.9.3\"" -DPACKAGE_BUGREPORT=\"simon@simon-cozens.org\" -DPACKAGE_URL=\"\" -DSTDC_HE
ADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".l
ibs/\" -DAPPKIT=1 -DHAVE_LUA_H=1 -DHAVE_LUALIB_H=1 -DHAVE_LAUXLIB_H=1 -DHAVE_LUACONF_H=1 -DSILE_PATH=\"/usr/local/share/sile\" -DSILE_LIB_PATH=\"/usr/local/lib/sile\" -I. -I/usr/local/Cellar/freetype/2.6_1/inclu
de/freetype2 -fmodules -g -O2 -MT macfonts_la-macfonts.lo -MD -MP -MF .deps/macfonts_la-macfonts.Tpo -c macfonts.m  -fno-common -DPIC -o .libs/macfonts_la-macfonts.o
macfonts.m:142:15: warning: assigning to 'char *' from 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
    font_path = lua_tostring(L, -1);
```

Still, it’s giving the same error.

**Comment by simoncozens** [comment]
I am a dummy. Please update and try again.

**Comment by deepakjois** [comment]
Works now.

**Comment by behdad** [comment]
Humm.  What's the fontconfig-related problem again?

**Comment by simoncozens** [comment]
$ fc-match 'Geeza Pro:style=Regular'
    DejaVuSans.ttf: "DejaVu Sans" "Book"

    $ fc-match 'GeezaPro:style=Regular'
    DejaVuSans.ttf: "DejaVu Sans" "Book"

    $ fc-match 'جيزة:style=Regular'
    GeezaPro.ttc: "جيزة" "Regular"

    $ fc-list | grep Geeza
    /System/Library/Fonts/GeezaPro.ttc: جيزة:style=Regular,標準體,Ordinær,Normal,Normaali,Regolare,レギュラー,일반체,Regulier,Обычный,常规体,عادي
    /System/Library/Fonts/GeezaPro.ttc: .Geeza Pro Interface:style=Bold
    /System/Library/Fonts/GeezaPro.ttc: .Geeza Pro Interface:style=Regular
    /System/Library/Fonts/GeezaPro.ttc: .Geeza PUA:style=Bold,粗體,Fed,Fett,Puolilihava,Gras,Grassetto,ボールド,볼드체,Vet,Fet,Negrito,Жирный,粗体,عريض,Negrita
    /System/Library/Fonts/GeezaPro.ttc: .Geeza PUA:style=Regular,標準體,Ordinær,Normal,Normaali,Regolare,レギュラー,일반체,Regulier,Обычный,常规体,عادي
    /System/Library/Fonts/GeezaPro.ttc: .Geeza Pro Interface:style=Light
    /System/Library/Fonts/GeezaPro.ttc: جيزة:style=Bold,粗體,Fed,Fett,Puolilihava,Gras,Grassetto,ボールド,볼드체,Vet,Fet,Negrito,Жирный,粗体,عريض,Negrita

**Comment by behdad** [comment]
> $ fc-list | grep Geeza /System/Library/Fonts/GeezaPro.ttc: جيزة:style=Regular,標準體,Ordinær,Normal,Normaali,Regolare,レギュラー,일반체,Regulier,Обычный,常规体,عادي

This was a problem with fontconfig at some point, since we were only listing Unicode-encoding names.  I reinstated Apple MacRoman names to handle cases like this.  Unfortunately that version is unreleased right now, blocked by me.  We'll get a release out in a few days. :|  sorry about the problem.

**Comment by behdad** [comment]
Actually, the fix is in 2.11.93 and 2.11.94 already.
